bindings/python/setup.py

```diff
@@ -49,78 +86,84 @@ def append_libbson_flags(module):
         orig_ldflags = os.environ.get('LDFLAGS', '')
         os.environ['LDFLAGS'] = ldflags + " " + orig_ldflags
 
-    # Ensure our Cython extension can dynamically link to vendored libbson
-    # - https://blog.krzyzanowskim.com/2018/12/05/rpath-what/
-    # - https://nehckl0.medium.com/creating-relocatable-linux-executables-by-setting-rpath-with-origin-45de573a2e98
-    if platform == "darwin":
-        module.extra_link_args += ["-rpath", "@loader_path"]
-    elif platform == 'linux':
-        module.extra_link_args += ["-Wl,-rpath,$ORIGIN"]
-
     # https://cython.readthedocs.io/en/latest/src/tutorial/external.html#dynamic-linking
-    lnames = lnames.split()
     # Strip whitespace to avoid weird linker failures on manylinux images
-    libnames = [lname.lstrip('-l').strip() for lname in lnames]
+    libnames = [lname.lstrip('-l').strip() for lname in lnames.split()]
     module.libraries.extend(libnames)
 
 
 def append_arrow_flags(module):
     import numpy as np
     import pyarrow as pa
 
-    module.include_dirs.append(np.get_include())
-    module.include_dirs.append(pa.get_include())
-    module.library_dirs.extend(pa.get_library_dirs())
+    if os.name == 'nt':
+        module.include_dirs.append(np.get_include())
+        module.include_dirs.append(pa.get_include())
+    else:
+        module.extra_compile_args.append("-isystem" + pa.get_include())
+        module.extra_compile_args.append("-isystem" + np.get_include())
+        # Arrow's manylinux{2010, 2014} binaries are built with gcc < 4.8 which predates CXX11 ABI
+        # - https://uwekorn.com/2019/09/15/how-we-build-apache-arrows-manylinux-wheels.html
+        # - https://arrow.apache.org/docs/python/extending.html#example
+        if "std=" not in os.environ.get("CXXFLAGS", ""):
+            module.extra_compile_args.append("-std=c++11")
+            module.extra_compile_args.append("-D_GLIBCXX_USE_CXX11_ABI=0")
 
     # Handle the arrow library files manually.
     # Alternative to using pyarrow.create_library_symlinks().
-    # You can use MONGO_ARROW_LIBDIR to explicitly set the location of the 
+    # You can use MONGO_LIBARROW_DIR to explicitly set the location of the 
     # arrow libraries (for instance in a conda build).
     # We first check for an unmodified path to the library,
     # then look for a library file with a version modifier, e.g. libarrow.600.dylib.
-    arrow_lib = os.environ.get('MONGO_ARROW_LIBDIR', pa.get_library_dirs()[0])
+    arrow_lib = os.environ.get('MONGO_LIBARROW_DIR', pa.get_library_dirs()[0])
     if platform == "darwin":
         exts = ['.dylib', '.*.dylib']
     elif platform == 'linux':
         exts = ['.so', '.so.*']
```

Comment:
Doesn't there need to be a windows branch here?

Reply:
No, adding a comment